src/simulation/LoadGenerator.cpp

```diff
@@ -700,12 +752,36 @@ LoadGenerator::TxMetrics::report()
 }
 
 TransactionFramePtr
-LoadGenerator::createTransactionFramePtr(TestAccountPtr from,
-                                         std::vector<Operation> ops,
-                                         LoadGenMode mode)
+LoadGenerator::createTransactionFramePtr(
+    TestAccountPtr from, std::vector<Operation> ops, LoadGenMode mode,
+    std::optional<uint32_t> maxGeneratedFeeRate)
 {
-    auto txf = from->tx(ops);
+    int fee = 0;
 
+    if (maxGeneratedFeeRate)
+    {
+        auto baseFee = mApp.getLedgerManager().getLastTxFee();
+        if (baseFee > *maxGeneratedFeeRate)
+        {
+            throw std::runtime_error(
```

Comment:
hmm, this isn't an internal bug, just bad parameters from the user, so no need to crash here (need to gracefully fail instead). Also, is this the right place for the check? We could fast fail in CommandHandler right after parsing maxGeneratedFeeRate (and return an appropriate error message to the user in retStr)

Reply:
True, moved the check to `CommandHandler`.